proposals/safe-fixed-size-buffers.md

```diff
@@ -0,0 +1,399 @@
+
+Indexing into a fixed-size buffer with a constant expression outside of the declared fixed-size buffer bounds is a compile time error.
+
+The *safe-to-escape* scope of the value will be equal to the *safe-to-escape* scope of the *primary_no_array_creation_expression*,
+just as it would if the backing data was accessed as a field.
+
+### Conversions
+
+A new conversion from expression will be added.
+
+There is an implicit conversion from expression representing a variable of a fixed-size buffer type to the following types:
+- ```System.Span<T>``` - the result of conversion is a span returned by *AsSpan* method.
+- ```System.ReadOnlySpan<T>``` - the result of conversion is a span returned by *AsReadOnlySpan* method.
+
+There is an implicit conversion from expression representing a value of a fixed-size buffer type to ```System.ReadOnlySpan<T>``` type.
+The result of conversion is a span returned by *AsReadOnlySpan* method.
+
+### List patterns
+
+[List patterns](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-11.0/list-patterns.md) will not be supported for instances
+of fixed-size buffer types. In order to be able to take advantage of list patterns on fixed-size buffer types, a target expression must be
+explicitly converted either to ```System.Span<T>```, or to ```System.ReadOnlySpan<T>``` type first. The reason is the fact that pattern evaluation 
+liberally makes copies of the target expression, and it would be impossible to slice the copy into ```System.Span<T>```, but intent of the user might
+be to get instances of ```System.Span<T>``` from slicing. Even if the user would be fine with getting instances of ```System.ReadOnlySpan<T>```
+from slicing, it wouldn't be ref safe to return (from a pattern) one created based on a compiler generated temp. Lifetime of temps is very short
+(a statement or an expression) and they are often reused. When a target expression is copied is unspecified, and, often, depends on complexity of
+the pattern and the evaluation strategy chosen by compiler. 
+
+### MemoryMarshal.CreateReadOnlySpan
+
+We might want to specially recognize ```System.Runtime.InteropServices.MemoryMarshal.CreateReadOnlySpan``` method and allow
+passing a reference to a readonly location as its first argument. This should allow users to define their own fixed-size buffer
+types. For example, in order to be able to share a type across multiple assemblies and also include custom APIs into its definition.
+This might also help framework to define our shared fixed-size buffer types.
+
+### Definite assignment checking
+
+Regular definite assignment rules are applicable to variables that have a fixed-size buffer type. 
+
+## Open design questions
+
+### Initializer
+
+Should we support initialization at declaration site with, perhaps, [collection literals](https://github.com/dotnet/csharplang/blob/main/proposals/collection-literals.md)?
+
+## Alternatives
+
+### Method or property group like approach in the language
```

Comment:
This alternative looks like Option 2, what's the difference?

Reply:
Starting with

> Safe fixed-size buffer accesses have their own classification (just like e.g. method groups and lambdas)

The option 2 didn't introduce any new classification. Values of fixed-size buffer types are classified as values of any other struct type.